pandas/core/algorithms.py

```diff
@@ -267,22 +267,84 @@ def match(to_match, values, na_sentinel=-1):
     return result
 
 
-def unique1d(values):
+def unique(values):
     """
-    Hash table-based unique
+    Hash table-based unique. uniques are returned in order
+    of appearance. This does NOT sort.
+
+    Parameters
+    ----------
+    values : 1d array-like
+
+    Returns
+    -------
+    unique values.
+    - If the input is a Categorical dtype, the return is a Categorical
+    - If the input is an Index, the return is an Index
+    - If the input is a Series/ndarray, the return will be an ndarray
```

Comment:
I think it should rather return an Array of the correct type instead of a Series (with a meaningless index in case of uniques), but that's a discussion for the other issue tracker :-) (and this consistently for all input object types / data types)